### ISSUE TITLE
Missing, misplaced or mis-named DynamoDB does not create actionable error message

### ISSUE THREAD

**Issue by bcsgh** [hashicorp/terraform#24785]
### Terraform Version

```
$ terraform version
Terraform v0.12.24
+ provider.aws v2.58.0
+ provider.template v2.1.2
```

### Terraform Configuration Files

```hcl
terraform {
  backend "s3" {
    bucket = "bucket"
    key    = "key"
    region = "us-west-2"

    dynamodb_table = "us-west-2-table"  ## NOTE: created in us-east-2 (b/c AWS UX glitch)
  }
}
```

### Debug Output

Not relevant (it bug is with the *non* debug output). 

### Expected Behavior

An error message to the effect of `table 'foo' in region 'bar' does not exist: existing tables in that region will be listed at https://aws.example.com/region/.bar` or `table not accessible` or `Table resource not found` or something that gives some clue as to what the user needs to fix.

### Actual Behavior

```
Error: Error locking destination state: Error acquiring the state lock: 2 errors occurred:
	* ResourceNotFoundException: Requested resource not found
	status code: 400, request id: SOME1IG2LONG3REQUEST4ID5I6THINK
	* ResourceNotFoundException: Requested resource not found
	status code: 400, request id: A0HASH9OR8NONCE8OF7SOME6KIND
```

### Steps to Reproduce

1. Set up remote locking
1. create the DynamoDB in the wrong region.
1. `terraform init`

### Additional Context

I know that this is prompted by user (i.e. me) error, but the bug I'm reporting is that for this very predictable case, the tool give no help at all for figuring out what the issues is

### References

What eventually led me to the underlining issue. 
https://groups.google.com/forum/#!topic/terraform-tool/HpN8iJ7NKRA

**Comment by bcsgh** [comment]
Note: when the s3 bucket is the issue, a nice error message is generated:

```
Error: Error inspecting states in the "s3" backend:
    S3 bucket does not exist.

The referenced S3 bucket must have been previously created. If the S3 bucket
was created within the last minute, please wait for a minute or two and try
again.

Error: NoSuchBucket: The specified bucket does not exist
	status code: 404, request id: ..., host id: ...
```

**Comment by gdavison** [comment]
This should be fixed by #33851
